### PR TITLE
storage: Use proper Patternfly markup for comboboxes

### LINKED ISSUE #8560: storage: NFS path dropdown is not visible
In the [usability testing of NFS](
https://github.com/cockpit-project/cockpit/wiki/Usability-test-NFSclient
), we identified that none of the four participants noticed the dropdown for the NFS path.

### PR DESCRIPTION
**Pull request by mvollmer**
- [x] #8833 

Fixes #8560

### CODE REVIEW

**Comment by mvollmer** [comment]
This also improves the server path dropdown content, see the relevant commit.

File: pkg/storaged/dialog.js [hunk 1]
```diff
@@ -287,12 +287,18 @@
...
             if (typeof choices == 'function') {
-                $.when(choices(get_field_values())).then(function (result) {
-                    if (result !== false)
-                        combobox_set_choices(name, result);
-                });
+                set_choices_counter += 1;
+                var count = set_choices_counter;
+                choices(get_field_values(),
+                        function (ch) {
+                            console.log("R", ch, count, set_choices_counter);
```
Comment (martinpitt) [inline pkg/storaged/dialog.js]: Debugging leftover?

Admittedly I don't understand this. `count` is defined to be `set_choices_counter`, and never gets changed. So how would `if (count == set_choices_counter)` ever be false? What is this counter meant to achieve?

File: pkg/storaged/nfs-details.jsx [hunk 2]
```diff
@@ -70,40 +70,27 @@ function nfs_busy_dialog(client,
 export function nfs_fstab_dialog(client, entry) {
 
     var server_to_check;
-    var server_check_deferred;
 
-    function remote_choices(vals) {
+    function remote_choices(vals, setter) {
+        console.log("M", vals);
```
Comment (martinpitt) [inline pkg/storaged/nfs-details.jsx]: debugging leftover

File: pkg/storaged/dialog.js (hunk 1 above)
Comment (mvollmer) [reply pkg/storaged/dialog.js]: > Debugging leftover?

Indeed!

> count is defined to be set_choices_counter, and never gets changed.

It is `set_choices_counter` that might get changed.  Every new call to `combobox_set_choices` should cancel all the background activity of all previous calls.  But that's hard, so we just ignore callback invocations that come too late.

File: pkg/storaged/nfs-details.jsx (hunk 2 above)
Comment (mvollmer) [reply pkg/storaged/nfs-details.jsx]: Removed

**Pull request closed without merging**